File: app/src/main/java/com/rajendra/sketchide/editor/palette/callers/ViewCaller.java
package com.rajendra.sketchide.editor.palette.callers;

import android.content.Context;
import android.view.View;

import com.rajendra.sketchide.managers.IdManager;
import com.rajendra.sketchide.utils.Constants;
import com.rajendra.sketchide.utils.DimensionUtil;

import java.util.Objects;

public class ViewCaller {
  public static void setId(View target, String value, Context context) {
    IdManager.addNewId(target, value);
  }

  public static void setLayoutWidth(View target, String value, Context context) {
    target.getLayoutParams().width = (int) DimensionUtil.parse(value, context);
    target.requestLayout();
  }

  public static void setLayoutHeight(View target, String value, Context context) {
    target.getLayoutParams().height = (int) DimensionUtil.parse(value, context);
    target.requestLayout();
  }

  public static void setElevation(View target, String value, Context context) {
    target.setElevation(DimensionUtil.parse(value, context));
  }

  public static void setAlpha(View target, String value, Context context) {
    target.setAlpha(Float.parseFloat(value));
  }

  public static void setRotation(View target, String value, Context context) {
    target.setRotation(Float.parseFloat(value));
  }

  public static void setRotationX(View target, String value, Context context) {
    target.setRotationX(Float.parseFloat(value));
  }

  public static void setRotationY(View target, String value, Context context) {
    target.setRotationY(Float.parseFloat(value));
  }

  public static void setTranslationX(View target, String value, Context context) {
    target.setTranslationX(DimensionUtil.parse(value, context));
  }

  public static void setTranslationY(View target, String value, Context context) {
    target.setTranslationY(DimensionUtil.parse(value, context));
  }

  public static void setTranslationZ(View target, String value, Context context) {
    target.setTranslationZ(DimensionUtil.parse(value, context));
  }

  public static void setScaleX(View target, String value, Context context) {
    target.setScaleX(Float.parseFloat(value));
  }

  public static void setScaleY(View target, String value, Context context) {
    target.setScaleY(Float.parseFloat(value));
  }

  public static void setPadding(View target, String value, Context context) {
    int pad = (int) DimensionUtil.parse(value, context);
    target.setPadding(pad, pad, pad, pad);
  }

  public static void setPaddingLeft(View target, String value, Context context) {
    int pad = (int) DimensionUtil.parse(value, context);
    target.setPadding(
      pad, target.getPaddingTop(), target.getPaddingRight(), target.getPaddingBottom());
  }

  public static void setPaddingRight(View target, String value, Context context) {
    int pad = (int) DimensionUtil.parse(value, context);
    target.setPadding(
      target.getPaddingLeft(), target.getPaddingTop(), pad, target.getPaddingBottom());
  }

  public static void setPaddingTop(View target, String value, Context context) {
    int pad = (int) DimensionUtil.parse(value, context);
    target.setPadding(
      target.getPaddingLeft(), pad, target.getPaddingRight(), target.getPaddingBottom());
  }

  public static void setPaddingBottom(View target, String value, Context context) {
    int pad = (int) DimensionUtil.parse(value, context);
    target.setPadding(
      target.getPaddingLeft(), target.getPaddingTop(), target.getPaddingRight(), pad);
  }

  public static void setEnabled(View target, String value, Context context) {
    target.setEnabled(value.equals("true"));
  }

  public static void setVisibility(View target, String value, Context context) {
    target.setVisibility(Objects.requireNonNull(Constants.visibilityMap.get(value)));
  }
}

File: app/src/main/java/com/rajendra/sketchide/editor/palette/callers/ViewCaller.kt
package com.rajendra.sketchide.editor.palette.callers

import android.content.Context
import android.graphics.Color
import android.graphics.drawable.ColorDrawable
import android.view.View
import com.rajendra.sketchide.managers.IdManager.addNewId
import com.rajendra.sketchide.utils.ArgumentUtil
import com.rajendra.sketchide.utils.Constants
import com.rajendra.sketchide.utils.DimensionUtil


object ViewCaller {
  fun setId(target: View?, value: String, context: Context) {
    addNewId(target!!, value)
  }

  fun setLayoutWidth(target: View, value: String, context: Context) {
    target.layoutParams.width = DimensionUtil.parse(value, context).toInt()
    target.requestLayout()
  }

  fun setLayoutHeight(target: View, value: String, context: Context) {
    target.layoutParams.height = DimensionUtil.parse(value, context).toInt()
    target.requestLayout()
  }

//  fun setBackground(target: View, value: String, context: Context?) {
//    if (ArgumentUtil.parseType(value, arrayOf<String>("color", "drawable")) == ArgumentUtil.COLOR
//    ) {
//      target.setBackgroundColor(Color.parseColor(value))
//    } else {
//      val name = value.replace("@drawable/", "")
//      target.background = DrawableManager.getDrawable(context, name)
//    }
//  }
//
//  fun setForeground(target: View, value: String, context: Context?) {
//    if (ArgumentUtil.parseType(value, arrayOf<String>("color", "drawable"))
//        .equals(ArgumentUtil.COLOR)
//    ) {
//      target.foreground = ColorDrawable(Color.parseColor(value))
//    } else {
//      val name = value.replace("@drawable/", "")
//      target.foreground = DrawableManager.getDrawable(context, name)
//    }
//  }

  fun setElevation(target: View, value: String, context: Context) {
    target.elevation = DimensionUtil.parse(value, context)
  }

  fun setAlpha(target: View, value: String, context: Context) {
    target.alpha = value.toFloat()
  }

  fun setRotation(target: View, value: String, context: Context) {
    target.rotation = value.toFloat()
  }

  fun setRotationX(target: View, value: String, context: Context) {
    target.rotationX = value.toFloat()
  }

  fun setRotationY(target: View, value: String, context: Context) {
    target.rotationY = value.toFloat()
  }

  fun setTranslationX(target: View, value: String, context: Context) {
    target.translationX = DimensionUtil.parse(value, context)
  }

  fun setTranslationY(target: View, value: String, context: Context) {
    target.translationY = DimensionUtil.parse(value, context)
  }

  fun setTranslationZ(target: View, value: String, context: Context) {
    target.translationZ = DimensionUtil.parse(value, context)
  }

  fun setScaleX(target: View, value: String, context: Context) {
    target.scaleX = value.toFloat()
  }

  fun setScaleY(target: View, value: String, context: Context) {
    target.scaleY = value.toFloat()
  }

  fun setPadding(target: View, value: String, context: Context) {
    val pad = DimensionUtil.parse(value, context).toInt()
    target.setPadding(pad, pad, pad, pad)
  }

  fun setPaddingLeft(target: View, value: String, context: Context) {
    val pad = DimensionUtil.parse(value, context).toInt()
    target.setPadding(
      pad, target.paddingTop, target.paddingRight, target.paddingBottom
    )
  }

  fun setPaddingRight(target: View, value: String, context: Context) {
    val pad = DimensionUtil.parse(value, context).toInt()
    target.setPadding(
      target.paddingLeft, target.paddingTop, pad, target.paddingBottom
    )
  }

  fun setPaddingTop(target: View, value: String, context: Context) {
    val pad = DimensionUtil.parse(value, context).toInt()
    target.setPadding(
      target.paddingLeft, pad, target.paddingRight, target.paddingBottom
    )
  }

  fun setPaddingBottom(target: View, value: String, context: Context) {
    val pad = DimensionUtil.parse(value, context).toInt()
    target.setPadding(
      target.paddingLeft, target.paddingTop, target.paddingRight, pad
    )
  }

  fun setEnabled(target: View, value: String, context: Context?) {
    if (value == "true") target.isEnabled = true
    else target.isEnabled = false
  }

  fun setVisibility(target: View, value: String, context: Context) {
    target.visibility = Constants.visibilityMap[value]!!
  }
}